### PR TITLE
Add functionality to convert file inputs to either json, csv or tsv.

### PR DESCRIPTION
**Pull request by john-thuo1**
### Contributor Checklist
- [✔️] This pull request is on a [separate branch](https://docs.github.com/en/get-started/quickstart/github-flow) and not the main branch.
### Description
This pull request adds functionality to convert data into CSV, TSV, and JSON formats. The changes include:
- Extension of the `convert_to_json` and `convert_to_csv_or_tsv` functions in `convert.py`.
- Updates to the CLI in `main.py` to support new conversion commands.
- Updates to the `get` function to support different file formats other that JSON.
#### Example Commands:
1. **From JSON to CSV**:
   ```bash
   scribe-data convert --lan French --data-type translations --input-file ./fli/French/translations.json --output-type csv --output-dir ./converted/
2. **From CSV to JSON**:
   ```bash
   scribe-data convert --lan French --data-type translations --input-file ./converted/French/translations.csv --output-type json --output-dir ./converted/
3. **From JSON to TSV**:
   ```bash
   scribe-data convert --lan French --data-type translations --input-file ./fli/French/translations.json --output-type tsv --output-dir ./converted/ 
4. **From TSV to JSON**:
   ```bash
   scribe-data convert --lan French --data-type translations --input-file ./converted/French/translations.tsv --output-type json --output-dir ./converted/ 

![image](https://github.com/user-attachments/assets/ce6ed679-faa1-40bd-a53b-ee94e2971a4b)

Tests for the convert methods have been implemented in `test_convert.py`

### Related issue

- Decouple convert from get in Scribe-Data CLI #213

### CODE REVIEW

**Review by mhmohona — APPROVED** [review]
I think it looks good.

**Comment by john-thuo1** [comment]
> It looks good. Great work @john-thuo1 ✨

Alright! I will look into the remaining tests for the functions then,

**Comment by andrewtavis** [comment]
Yes that'd be great, @john-thuo1! I'll review once those are in :)

**Comment by andrewtavis** [comment]
Checking in here, @john-thuo1 :) Do you want to do the tests in a separate PR, or continue on this one? Whatever would be best for you 😊

**Comment by john-thuo1** [comment]
> Checking in here, @john-thuo1 :) Do you want to do the tests in a separate PR, or continue on this one? Whatever would be best for you 😊

I had some challenges figuring out how to mock the input file, output directory, and output file and patching some functionalities, hence the delay, but it is all good now!

I have finished the tests though. You can proceed with the review @andrewtavis

**Comment by andrewtavis** [comment]
Sorry for asking, @john-thuo1, but would you be able to fix the tests given the changes that went through to the project? I'll review directly after that. I can also look into this if you'd prefer :)

**Comment by john-thuo1** [comment]
> Sorry for asking, @john-thuo1, but would you be able to fix the tests given the changes that went through to the project? I'll review directly after that. I can also look into this if you'd prefer :)

I have just pulled the changes. Going through the failing tests!

**Comment by andrewtavis** [comment]
Thanks, @john-thuo1!

**Comment by andrewtavis** [comment]
Merged the emoji work in as well to get rid of a merge conflict for you, @john-thuo1! Let me know if there's anything else I can do to support!

**Comment by andrewtavis** [comment]
I'll take a look at this, @john-thuo1! Thanks so much :)

**Comment by andrewtavis** [comment]
Test is fine now, @john-thuo1, at least locally. I fixed it before and there was a change overwriting the fix. Checking the rest now 😊

File: src/scribe_data/cli/main.py [hunk 1]
```diff
@@ -224,7 +259,9 @@ def main() -> None:
         return
 
     if args.command in ["list", "l"]:
-        list_wrapper(args.language, args.data_type, args.all)
+        list_wrapper(
+            language=args.language, data_type=args.data_type, all_bool=args.all
```
Comment (andrewtavis) [inline src/scribe_data/cli/main.py:263]: Thanks for adding in the args! Really appreciate explicit function calls to make the functionality more clear.

File: src/scribe_data/cli/main.py [hunk 2]
```diff
@@ -245,21 +282,14 @@ def main() -> None:
...
-            )
-
-        elif args.output_type == "sqlite":
-            convert_to_sqlite(
-                args.language,
-                args.data_type,
-                args.output_dir,
-                args.overwrite,
-            )
+        convert(
```
Comment (andrewtavis) [inline src/scribe_data/cli/main.py:285]: Wonderful :)

Comment (andrewtavis) [reply src/scribe_data/cli/main.py:285]: Functioning just like the others now 😊 I'll do a rename to `convert_wrapper` for consistency and so people know this functions conditionally to call sub functions. More of a switcher, but then `total_switcher` sounds a bit weird.

**Comment by john-thuo1** [comment]
> Test is fine now, @john-thuo1, at least locally. I fixed it before and there was a change overwriting the fix. Checking the rest now 😊

It seems the test get emoji never calls `subprocess.run()`  . Is this sth you are observing as well? If it is the case would this alternative test case suffice ?
```
    @patch("scribe_data.cli.get.Path") 
    @patch("scribe_data.cli.get.generate_emoji") 
    def test_get_emoji_keywords(self, mock_generate_emoji, mock_path):
        mock_path_instance = mock_path.return_value  
        mock_path_instance.mkdir.return_value = None  
        mock_path_instance.exists.return_value = True 
        mock_path_instance.__str__.return_value = "scribe_data_json_export"
 
        get_data(language="English", data_type="emoji-keywords")

        self.assertTrue(mock_generate_emoji.called)

        mock_generate_emoji.assert_called_with(language="English", output_dir="scribe_data_json_export") ```

**Review by andrewtavis — APPROVED** [review]
Really really nice work here, @john-thuo1 :) Some tests needed to be commented out as I'd like to get this in, but overall this is a monumental addition to the project 😊 Really appreciate all the hard work and care you put into this!

**Comment by john-thuo1** [comment]
> Really really nice work here, @john-thuo1 :) Some tests needed to be commented out as I'd like to get this in, but overall this is a monumental addition to the project 😊 Really appreciate all the hard work and care you put into this!

@andrewtavis  @mhmohona  Thank you. The numerous back-and forths were very helpful in grasping the nuances of the task.

**Comment by andrewtavis** [comment]
Thank you, @john-thuo1! You were a perfect person to work through this :)

**Pull request merged**